### DISCUSSION TITLE
Debugging of feature files

### DISCUSSION THREAD

**Discussion by laeubi** [cucumber/cucumber-eclipse#441] [Ideas]
Currently it is possible to place breakpoints in the java code, but that's sometimes a bit to deep into the technical details. There is also the drawback that nearly no context information is available (e.g. one has to guess what step/scenario is currently executing.

Because of this, I have started an approach to make feature-files directly usable in the debugger, the following features are currently supported:

- placing a break-point in scenarios and steps
- if a break point is hit the cucumber-run is suspended and one gets a simple stacktrace of `Feature > Scenario > Step`
- in the variables section the current parsed parameter values of the current test step are displayed

Here is an image that shows the debugger stopped at a break-point in the `date_calculator.feature` on line 7

![grafik](https://user-images.githubusercontent.com/1331477/133484061-92340707-6198-4fce-8ed3-a7478964ad3b.png)

Let me know what you think about that feature, if it is useful for you, what other options/information should be displayed or could be required to see whats going on.

**Comment by nitind** [comment]
is this using JSR45 in some way?

**Reply by laeubi** [reply]
Not that I'm aware of it is implemented with the Eclipse Debugger Framework.
